web/src/pages/Dashboard.tsx

```diff
@@ -0,0 +1,282 @@
+      }
+      onRemoveCity(city.id);
+    }
+  };
+
+  return (
+    <SimpleGrid minChildWidth="25%" spacing={[1, 3]}>
+      {props.totalSelectableCities.map((city) => (
+        <Checkbox
+          size="md"
+          disabled={props.isUpdating}
+          key={city.id}
+          zIndex={10}
+          colorScheme="purple"
+          isChecked={
+            selectedCities.map((c) => c.id).includes(city.id) ? true : false
+          }
+          onChange={(e) => setCheckItem(city, e.target.checked)}
+        >
+          <Flex
+            justifyContent="left"
+            alignItems="center"
+            gap={1}
+            color={getCheckedFontColor(city)}
+          >
+            <CityNameConatiner city={city} />
+          </Flex>
+        </Checkbox>
+      ))}
+    </SimpleGrid>
+  );
+};
+
+const StatsWrapper = () => {
+  const [selectedCities] = useRecoilState(selectedCitiesFromRecoil);
+  const [isUpdating] = useRecoilState(isUpdatingCities);
+  const [lastSelectedCityId, setLastSelectedCityId] =
+    useRecoilState(selectedCity);
+  const [totalSelectableCities, setTotalSelectableCities] =
+    React.useState(selectedCities);
+
+  React.useEffect(() => {
+    const selectableCityIds = selectableCitiesData.map((c) => c.id);
+    const unknownSelectedCities = selectedCities.filter(
+      (c) => !selectableCityIds.includes(c.id)
+    );
+    setTotalSelectableCities([
+      ...selectableCitiesData,
+      ...unknownSelectedCities,
+    ]);
+    if (lastSelectedCityId === -1 && selectedCities.length > 0) {
+      setLastSelectedCityId(selectedCities[0].id);
+    }
+  }, [selectedCities, lastSelectedCityId, setLastSelectedCityId]);
+
+  return (
+    <>
+      <Stack spacing={4}>
+        <Stack spacing={3}>
+          <Stack spacing={2}>
+            <Heading fontSize="md">Locations</Heading>
+            <Text overflowWrap="break-word">
+              Select cities to add to the dataset
+            </Text>
+          </Stack>
+        </Stack>
+        <Tooltip
+          isDisabled={!isUpdating}
+          label="Updating city list"
+          hasArrow
+          placement="top"
+          zIndex={5}
+        >
+          <SelectCityCheckbox
+            isUpdating={isUpdating}
+            totalSelectableCities={totalSelectableCities}
+          />
+        </Tooltip>
+        <br />
+      </Stack>
+      <RealtimeChart />
+    </>
+  );
+};
+
+export const NotificationsMap = () => {
+  const { connected, initialized } = useConnectionState();
+  const enabled = useRecoilValue(simulatorEnabled);
+  useSimulationMonitor(enabled && connected && initialized);
+  useSimulator(enabled && connected && initialized);
+  const [isSmallScreen] = useMediaQuery("(max-width: 640px)");
+
+  if (!connected) {
+    window.location.href = "/";
+  }
+
+  let inner;
+  if (!initialized) {
+    inner = <SetupDatabaseButton />;
+  } else if (!enabled) {
+    inner = <EnableSimulatorButton />;
+  } else {
+    inner = <StatsWrapper />;
+  }
```

Comment:
nit: You should also give a more appropriate name than `inner`.

```suggestion
  let inner = <StatsWrapper />;
  if (!initialized) {
    inner = <SetupDatabaseButton />;
  } else if (!enabled) {
    inner = <EnableSimulatorButton />;
  }
```

Reply:
It already existed but yes can rename better !